controllers/components/trustyai/trustyai_controller.go

```diff
@@ -56,7 +58,22 @@ func (s *componentHandler) NewComponentReconciler(ctx context.Context, mgr ctrl.
 			reconciler.WithPredicates(
 				component.ForLabel(labels.ODH.Component(LegacyComponentName), labels.True)),
 		).
+		Watches(
+			&extv1.CustomResourceDefinition{},
+			reconciler.WithEventHandler(
+				handlers.ToNamed(componentApi.TrustyAIInstanceName)),
+			reconciler.WithPredicates(predicate.Or(
```

Comment:
can't this be merged with the CustomResourceDefinition watch above ?

Reply:
@lburgazzoli do you mean to just put it above EventHandler? Other controllers have the same order as here.

Reply:
The code has changed in the meantime, however the predicates now seems incorrect as, if I'm not wrong,  it would trigger a reconcile if any of the following conditions matches:
- a CRD belongs to `kserve`
- a CRD belongs to `model-mesh`
- a CRD names `inferenceservices.serving.kserve.io` is created

However:
- it does not take into account nay trusty ai specific CRD
- any CRD from `kserve`  or `model-mesh` would trigger a reconcile

I would expect to react on:
- Trusty AI CRDs
- Any platform CRD whose name is  `inferenceservices.serving.kserve.io`

Reply:
Oh, you are right
```go
			reconciler.WithPredicates(
				component.ForLabel(labels.ODH.Component(LegacyComponentName), labels.True)),
```
was removed in this PR. Should be put back, with the rest I agree too.

Reply:
yes